scrapy/crawler.py

```diff
@@ -221,6 +227,11 @@ def join(self):
         while self._active:
             yield defer.DeferredList(self._active)
 
+    def is_crawlers_has_spider(self):
+        return reduce(lambda x,y: x and y,
+                      self.crawlers_has_spiders,
+                      True)
```

Comment:
`any(self.crawlers_has_spiders)`?

Reply:
as @VMRuiz noted, it is actually `all(...)`, not `any(...)`, which is an additional argument to change it, as the code is not obvious :P